cpp/src/arrow/flight/server.cc

```diff
@@ -111,7 +115,11 @@ class SerializationTraits<IpcPayload> {
     }
 
     // 2 bytes for body tag
-    total_size += 2 + WireFormatLite::LengthDelimitedSize(static_cast<size_t>(body_size));
+    // Only written when there are body buffers
+    if (msg.body_length > 0) {
+      total_size +=
+          2 + WireFormatLite::LengthDelimitedSize(static_cast<size_t>(body_size));
+    }
```

Comment:
OK. We should probably open an issue to investigate this issue further. If a server were to serialize the FlightData using an actual protobuf library (instead of writing the protocol by hand) it could write a length-0 body (e.g. if the body was `b''` in Python). So that needs to not break any clients

Reply:
Filed https://issues.apache.org/jira/browse/ARROW-4419.